pkg/api/merge.go

```diff
@@ -0,0 +1,368 @@
+package api
+
+// TODO: needs unit tests
```

Comment:
Follow-up issue

Reply:
This is one of the most critical parts of code, I think we'd be wise to spend some time throwing in a unit test today to make sure we're performing how we expect.